lib/modules/manager/bazel-module/rules-img.ts

```diff
@@ -0,0 +1,88 @@
+import { z } from 'zod';
+import { DockerDatasource } from '../../datasource/docker';
+import type { PackageDependency } from '../types';
+import { RepoRuleCallFragment, StringFragment } from './parser/fragments';
+
+export const RulesImgPullCallToDep = RepoRuleCallFragment.extend({
+  children: z.object({
+    name: StringFragment,
+    repository: StringFragment,
+    registry: StringFragment.optional(),
+    tag: StringFragment.optional(),
+    digest: StringFragment.optional(),
+  }),
+}).transform(
+  ({
+    rawString,
+    functionName,
+    children: { name, repository, registry, tag, digest },
+  }): PackageDependency => {
+    // Note: Validation that this is a rules_img pull call is done in transformRulesImgCalls
+
+    // Construct the package name
+    let packageName = repository.value;
+    if (registry?.value) {
+      packageName = `${registry.value}/${repository.value}`;
+    }
+
+    return {
+      datasource: DockerDatasource.id,
+      depType: 'rules_img_pull',
+      depName: name.value,
+      packageName,
+      currentValue: tag?.value,
+      currentDigest: digest?.value,
+      registryUrls: registry?.value
+        ? [`https://${registry.value}`]
+        : [`https://index.docker.io`],
```

Comment:
If custom registry is found assign it. No need to add the docker hub url as fallback as you are already setting datasource=docker

Reply:
resolved via the last change